ui/src/lib.rs

```diff
@@ -1250,6 +1282,24 @@ enum Interrupt {
     IntervalElapsed,
 }
 
+impl Interrupt {
+    fn char_press(&self) -> Option<char> {
+        if let Self::KeyPressed(event) = &self {
+            if let KeyCode::Char(c) = event.code {
+                if c.is_alphanumeric() || c == '\'' {
+                    Some(c)
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+}
```

Comment:
please simplify by collapsing, use `return Some()` and None as default return end of function